airflow/www/api/experimental/endpoints.py

```diff
@@ -172,6 +173,42 @@ def task_instance_info(dag_id, execution_date, task_id):
     return jsonify(fields)
 
 
+@api_experimental.route(
+    '/dags/<string:dag_id>/dag_runs/<string:execution_date>/status',
+    methods=['GET'])
+@requires_authentication
+def dag_run_status(dag_id, execution_date):
```

Comment:
I would only defined this function in the `www_rbac` one. Since we don't use this code in the tests, I would argue that it isn't part of the PR. Also only the rbac code is tested. What do you think?

Reply:
I'm fine with that. If someone needs this for a non-RBAC use-case they're welcome to copy it in another PR